### PR TITLE
prevent external hostname pullthrough + bump digest cache size

### PR DESCRIPTION
**Pull request by dmage**
Port of #161 to master.

### CODE REVIEW

File: pkg/imagestream/imagestream.go [hunk 1]
```diff
@@ -241,21 +241,31 @@ func (is *imageStream) Exists(ctx context.Context) (bool, *rerrors.Error) {
...
+	var localNames []string
+
 	local, err := imageapi.ParseDockerImageReference(stream.Status.DockerImageRepository)
 	if err != nil {
-		return "", rerrors.NewError(
-			ErrImageStreamUnknownErrorCode,
-			fmt.Sprintf("localRegistry: unable to parse reference %q", stream.Status.DockerImageRepository),
-			err,
-		)
+		context.GetLogger(ctx).Warnf("localRegistry: unable to parse dockerImageRepository %q", stream.Status.DockerImageRepository)
```
Comment (bparees) [inline pkg/imagestream/imagestream.go]: this needs to be dcontext in this branch

File: pkg/imagestream/imagestream.go [hunk 2]
```diff
@@ -241,21 +241,31 @@ func (is *imageStream) Exists(ctx context.Context) (bool, *rerrors.Error) {
...
+	}
+	if len(local.Registry) != 0 {
+		localNames = append(localNames, local.Registry)
 	}
-	return local.Registry, nil
+
+	public, err := imageapi.ParseDockerImageReference(stream.Status.PublicDockerImageRepository)
+	if err != nil {
+		context.GetLogger(ctx).Warnf("localRegistry: unable to parse publicDockerImageRepository %q", stream.Status.DockerImageRepository)
```
Comment (bparees) [inline pkg/imagestream/imagestream.go]: same here: this needs to be dcontext in this branch

File: pkg/imagestream/imagestream.go (hunk 1 above)
Comment (dmage) [reply pkg/imagestream/imagestream.go]: thanks

**Pull request merged**